src/sentry/static/sentry/app/views/asyncView.jsx

```diff
@@ -1,21 +1,82 @@
 import DocumentTitle from 'react-document-title';
+import idx from 'idx';
+import Modal from 'react-bootstrap/lib/Modal';
 import React from 'react';
 
 import AsyncComponent from '../components/asyncComponent';
+import Button from '../components/buttons/button';
+import ConfigStore from '../stores/configStore';
+import {ApiForm, PasswordField} from '../components/forms';
+import {t} from '../locale';
 
-class AsyncView extends AsyncComponent {
-  constructor(...args) {
-    super(...args);
-  }
-
+export default class AsyncView extends AsyncComponent {
   getTitle() {
     return 'Sentry';
   }
+
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      needsVerification: false,
+    };
+  }
+
+  handleError(error, ...args) {
+    if (idx(error, _ => _.responseJSON.needsSuperuser)) {
```

Comment:
It's probably better to handle this outside of AsyncView so that it applies to all API requests.

I'll open a PR to refactor this when #6727 lands.

Reply:
@billyvg it'd need a singleton or some other complex mechanism to handle it in individual components, that was my main reason for this

e.g. you make 3 api requests, N return this response, what happens you auth?

Reply:
Yeah that shouldn't be a huge change, the sudo modal currently uses a store. We can have the store queue up all requests that failed and try them all when you auth.